app/models/discussion_reader.rb

```diff
@@ -53,20 +68,41 @@ def discussion_reader_volume
   end
 
   private
+  def serialize_read_ranges(ranges)
+    # could merge adjancent ranges in the future
+    self.read_sequence_id_ranges = ranges.sort_by{|r| r.first}.map{|r| [r.first,r.last].join(',')}.join(' ')
```

Comment:
So the data stored here is `1,4 6,10 15,20`, which means you've missed sequence 5 and 11-14?

Reply:
```
ranges.sort_by(&:first)
```

Reply:
What happens if there's a single item 'read'? (and is that even possible?)... it looks like we do `5,5`; do we handle that edge case properly?

Reply:
yes that's handled